Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseTelemetryProcessorTests.cs

```diff
@@ -694,5 +710,300 @@ public void QuickPulseTelemetryProcessorDoesNotCollectFullTelemetryItemsWhenSwit
+
+            // ACT
+            var dependencyShort = new DependencyTelemetry(
+                new string('c', MaxFieldLength),
+                new string('c', MaxFieldLength),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false) { Context = { InstrumentationKey = instrumentationKey } };
+
+            var dependencyLong = new DependencyTelemetry(
+                new string('c', MaxFieldLength + 1),
+                new string('c', MaxFieldLength + 1),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false) { Context = { InstrumentationKey = instrumentationKey } };
+
+            // process in the opposite order to allow for an easier validation order
+            telemetryProcessor.Process(dependencyLong);
+            telemetryProcessor.Process(dependencyShort);
+
+            // ASSERT
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[0]).CommandName,
+                dependencyShort.CommandName);
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[1]).CommandName,
+                dependencyLong.CommandName.Substring(0, MaxFieldLength));
+        }
+
+        [TestMethod]
+        public void QuickPulseTelemetryProcessorTruncatesLongFullDependencyTelemetryItemName()
+        {
+            // ARRANGE
+            var accumulatorManager = new QuickPulseDataAccumulatorManager();
+            var telemetryProcessor = new QuickPulseTelemetryProcessor(new SimpleTelemetryProcessorSpy());
+            var instrumentationKey = "some ikey";
+            ((IQuickPulseTelemetryProcessor)telemetryProcessor).StartCollection(
+                accumulatorManager,
+                new Uri("http://microsoft.com"),
+                new TelemetryConfiguration() { InstrumentationKey = instrumentationKey });
+
+            // ACT
+            var dependencyShort = new DependencyTelemetry(
+                new string('c', MaxFieldLength),
+                new string('c', MaxFieldLength),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false)
+            { Context = { InstrumentationKey = instrumentationKey } };
+
+            var dependencyLong = new DependencyTelemetry(
+                new string('c', MaxFieldLength + 1),
+                new string('c', MaxFieldLength + 1),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false)
+            { Context = { InstrumentationKey = instrumentationKey } };
+
+            // process in the opposite order to allow for an easier validation order
+            telemetryProcessor.Process(dependencyLong);
+            telemetryProcessor.Process(dependencyShort);
```

Comment:
Probably the cleanest I can think of that doesn't involve an ugly helper method with out parameters would be to use the [TestInitialize()] attribute on a test setup method which creates and sets the variables you need each time.  The test framework will run that attributed init method before it runs each test so you could make the variables you need class instance level and then trust the test framework to always run that method before running each test.
If however your tests are only a few out of many in this file then it won't make sense to do that because there is no reason to have those variables re-created for tests which don't use them.